### ISSUE TITLE
MOT17-13-FRCNN 0.0%  NaN 44 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  NaN        3156

### ISSUE THREAD

**Issue by SmallGGgg** [ifzhang/ByteTrack#390]
我在测试的时候遇到了下面的问题，请问是代码的问题还是数据集脚本的问题？
I encountered the following problems during testing. Is it the code or the data set script?
 0%|          | 0/2652 [00:00<?, ?it/s]/media/cvlab1045/D/danconda/envs/ngf38/lib/python3.8/site-packages/torch/functional.py:568: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  ../aten/src/ATen/native/TensorShape.cpp:2228.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
100%|#########9| 2651/2652 [09:29<00:00,  4.62it/s]2024-03-04 19:42:57 | INFO     | yolox.evaluators.mot_evaluator:42 - save results to ./ap87.4/huatuceshi/lxf_test1/track_results/MOT17-13-FRCNN.txt
100%|##########| 2652/2652 [09:30<00:00,  4.65it/s]
2024-03-04 19:42:57 | INFO     | yolox.evaluators.mot_evaluator:689 - Evaluate in main process...
2024-03-04 19:42:57 | INFO     | __main__:248 - 
Average forward time: 212.86 ms, Average track time: 0.02 ms, Average inference time: 212.87 ms

gt_type _val_half
gt_files ['/media/cvlab1045/E/NGF/ByteTrack-main/datasets/mot/train/MOT17-09-FRCNN/gt/gt_val_half.txt', '/media/cvlab1045/E/NGF/ByteTrack-main/datasets/mot/train/MOT17-10-FRCNN/gt/gt_val_half.txt', '/media/cvlab1045/E/NGF/ByteTrack-main/datasets/mot/train/MOT17-11-FRCNN/gt/gt_val_half.txt', '/media/cvlab1045/E/NGF/ByteTrack-main/datasets/mot/train/MOT17-04-FRCNN/gt/gt_val_half.txt', '/media/cvlab1045/E/NGF/ByteTrack-main/datasets/mot/train/MOT17-05-FRCNN/gt/gt_val_half.txt', '/media/cvlab1045/E/NGF/ByteTrack-main/datasets/mot/train/MOT17-02-FRCNN/gt/gt_val_half.txt', '/media/cvlab1045/E/NGF/ByteTrack-main/datasets/mot/train/MOT17-13-FRCNN/gt/gt_val_half.txt']
2024-03-04 19:42:57 | INFO     | __main__:265 - Found 7 groundtruths and 1 test files.
2024-03-04 19:42:57 | INFO     | __main__:266 - Available LAP solvers ['lap', 'scipy']
2024-03-04 19:42:57 | INFO     | __main__:267 - Default LAP solver 'lap'
2024-03-04 19:42:57 | INFO     | __main__:268 - Loading files.
2024-03-04 19:42:58 | INFO     | __main__:134 - Comparing MOT17-13-FRCNN...
2024-03-04 19:42:58 | INFO     | __main__:276 - Running metrics
               Rcll Prcn GT   MT   PT     ML   FP     FN  IDs   FM MOTA MOTP num_objects
MOT17-13-FRCNN 0.0%  NaN 44 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  NaN        3156
OVERALL        0.0%  NaN 44 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  NaN        3156
               IDF1 IDP  IDR Rcll Prcn GT MT PT ML FP   FN IDs  FM MOTA MOTP IDt IDa IDm num_objects
MOT17-13-FRCNN 0.0% NaN 0.0% 0.0%  NaN 44  0  0 44  0 3156   0   0 0.0%  NaN   0   0   0        3156
OVERALL        0.0% NaN 0.0% 0.0%  NaN 44  0  0 44  0 3156   0   0 0.0%  NaN   0   0   0        3156
2024-03-04 19:42:58 | INFO     | __main__:301 - Completed

**Comment by xinranzero** [comment]
I have met the same question. Only the txt file corresponding to the last file name is saved in the "track_result" folder, and it is empty. This shows that the results of inference are not properly preserved. I am trying to solve this problem, please let me know if you have solved it. Many thanks!!!!!！

**Comment by SmallGGgg** [comment]
> I have met the same question. Only the txt file corresponding to the last file name is saved in the "track_result" folder, and it is empty. This shows that the results of inference are not properly preserved. I am trying to solve this problem, please let me know if you have solved it. Many thanks!!!!!！

I haven't solved it yet, if you solve it, please teach me, thank you very much!

**Comment by SmallGGgg** [comment]
> I have met the same question. Only the txt file corresponding to the last file name is saved in the "track_result" folder, and it is empty. This shows that the results of inference are not properly preserved. I am trying to solve this problem, please let me know if you have solved it. Many thanks!!!!!！

100%|#########9| 2651/2652 [05:01<00:00,  9.05it/s]2024-03-06 15:37:38 | INFO     | yolox.evaluators.mot_evaluator:42 - save results to ./me_Fusionv3_ablation/lxf_test/track_results/MOT17-13-FRCNN.txt
100%|##########| 2652/2652 [05:01<00:00,  8.79it/s]
2024-03-06 15:37:38 | INFO     | yolox.evaluators.mot_evaluator:658 - Evaluate in main process...
gt_type _val_half
gt_files []
2024-03-06 15:37:38 | INFO     | __main__:248 - 
Average forward time: 111.67 ms, Average track time: 0.02 ms, Average inference time: 111.69 ms

2024-03-06 15:37:38 | INFO     | __main__:265 - Found 0 groundtruths and 1 test files.
2024-03-06 15:37:38 | INFO     | __main__:266 - Available LAP solvers ['lap', 'scipy']
2024-03-06 15:37:38 | INFO     | __main__:267 - Default LAP solver 'lap'
2024-03-06 15:37:38 | INFO     | __main__:268 - Loading files.
2024-03-06 15:37:38 | WARNING  | __main__:138 - No ground truth for MOT17-13-FRCNN, skipping.
2024-03-06 15:37:38 | INFO     | __main__:276 - Running metrics
        Rcll Prcn GT  MT  PT  ML  FP  FN IDs  FM MOTA MOTP num_objects
OVERALL  NaN  NaN  0 NaN NaN NaN NaN NaN NaN NaN  NaN  NaN           0
        IDF1 IDP IDR Rcll Prcn GT MT PT ML FP FN IDs  FM MOTA MOTP IDt IDa IDm num_objects
OVERALL  NaN NaN NaN  NaN  NaN  0  0  0  0  0  0   0   0  NaN  NaN   0   0   0           0
2024-03-06 15:37:38 | INFO     | __main__:301 - Completed

I have a new problem

**Comment by xinranzero** [comment]
For your new problem:    Please check that your gt.txt file is saved in the correct location, as gt_files [] is displayed as empty in your result print.

**Comment by xinranzero** [comment]
Originally I could only get an empty prediction with the last file name. Then I tried the following:

1. I carefully adjusted the format of the `json` file to keep it consistent with the official one. After doing this I was able to get all the prediction result files (even though they were empty) during the detection phase. In this case, the model can work properly, but no target can be detected.
2. Then I tried to use the officially trained weight file for detection, and the result showed that it could produce some output. So maybe the officially trained weights don't match my dataset.
3. I checked the `train_log.txt` file and found that there were many warnings in my training, which indicated that the depth and width values used were inconsistent with the model I used. I changed the depth and width values and retrained for 4 epoches, and some results were produced (although the results were not ideal).
4. Increase the learning rate to speed up training.

Now I'm trying larger epoches of training, wish me luck!

**Comment by mtmyyy** [comment]
mot17官方数据集test貌似没有gt，所以没法计算评价指标

**Comment by xinranzero** [comment]
You can use the first half of the training set for training and the second half for validation. Run the officially provided command line 
```
cd <ByteTrack_HOME>
python3 tools/convert_mot17_to_coco.py 
```
will give you the gt_val_half.txt file. By the way, my previous failure to detect the result was a problem with the dataset, I replaced a dataset that was easier to detect the objects, and then successfully output files with the detection results.

**Comment by SmallGGgg** [comment]
> You can use the first half of the training set for training and the second half for validation. Run the officially provided command line
> 
> ```
> cd <ByteTrack_HOME>
> python3 tools/convert_mot17_to_coco.py 
> ```
> 
> will give you the gt_val_half.txt file. By the way, my previous failure to detect the result was a problem with the dataset, I replaced a dataset that was easier to detect the objects, and then successfully output files with the detection results.

Thank you very much！ I will try, if there are still problems, I will continue to ask you ~

**Comment by SmallGGgg** [comment]
> mot17官方数据集test貌似没有gt，所以没法计算评价指标

是的，但我是在验证集上测试的
